crates/biome_js_analyze/src/lint/nursery/no_setup_props_reactivity_loss.rs

```diff
@@ -0,0 +1,386 @@
+    ///       console.log(props.count)
+    ///     })
+    ///     return () => h('div', props.count)
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// // Destructuring inside callbacks is OK
+    /// export default {
+    ///   setup(props) {
+    ///     watch(() => props.count, () => {
+    ///       const { count } = props // OK inside callback
+    ///       console.log(count)
+    ///     })
+    ///     return () => {
+    ///       const { count } = props // OK inside render function
+    ///       return h('div', count)
+    ///     }
+    ///   }
+    /// }
+    /// ```
+    ///
+    pub NoSetupPropsReactivityLoss {
+        version: "next",
+        name: "noSetupPropsReactivityLoss",
+        language: "js",
+        recommended: false,
+    }
+}
+
+/// Represents the different types of reactivity loss violations that can occur.
+pub enum RuleState {
+    /// Props destructured directly in setup function parameters.
+    DestructuredParameter(TextRange),
+    /// Props destructured in variable assignments within the setup function body.
+    DestructuredAssignment(TextRange),
+}
+
+impl Rule for NoSetupPropsReactivityLoss {
+    type Query = Semantic<JsExportDefaultExpressionClause>;
+    type State = RuleState;
+    type Signals = Vec<Self::State>;
+    type Options = NoSetupPropsReactivityLossOptions;
+
+    /// Analyzes Vue component exports to detect props reactivity loss patterns.
+    ///
+    /// This method examines the default export of a Vue component, specifically
+    /// looking for setup methods that might cause props to lose reactivity through
+    /// destructuring or improper usage patterns.
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let export_clause = ctx.query();
+        let mut signals = Vec::new();
+
+        if let Ok(expression) = export_clause.expression()
+            && let Some(obj_expr) = expression.as_js_object_expression()
+            && let Some(setup_method) = find_setup_method(obj_expr)
+        {
+            check_setup_method(&setup_method, &mut signals);
+        }
+
+        signals
+    }
+
+    /// Creates diagnostic messages for detected reactivity loss violations.
+    ///
+    /// Generates appropriate error messages and helpful notes based on the type
+    /// of violation detected (parameter destructuring vs assignment destructuring).
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        match state {
+            RuleState::DestructuredParameter(range) => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    *range,
+                    markup! {
+                        "Destructuring props in setup function parameters causes reactivity loss."
+                    },
+                )
+                .note(markup! {
+                    "Use the props object directly to maintain reactivity: `props.propertyName`"
```

Comment:
This diagnostic doesn't contain the first pillar. We need to message error first, then explain why 

https://biomejs.dev/linter/#rule-pillars